### PR TITLE
small fix in plot_energy_hist() method of EventList class

### PR DESCRIPTION
**Pull request by cosimoNigro**
Hi all, running this simple snippet
```python
import numpy as np
import astropy.units as u
from gammapy.data import EventList

filename = '$GAMMAPY_EXTRA/datasets/fermi_2fhl/2fhl_events.fits.gz'
events = EventList.read(filename)
events.plot_energy_hist()
plt.show()
```

was triggering this issue
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/afs/ifh.de/group/amanda/scratch/cosimo/miniconda2/lib/python2.7/site-packages/gammapy/data/event_list.py", line 535, in plot_energy_hist
    emin = np.min(self['ENERGY'].quantity)
TypeError: 'EventList' object has no attribute '__getitem__'
```
there was a problem with `emin` and `emax` and the use of `spec = CountsSpectrum(energy=ebounds)` was outdated (I guess).
This is what I get after my fix
figure_1

### CODE REVIEW

**Comment by cdeil** [comment]
@cosimoNigro - Thanks!

Please always try to add a "regression test" when you fix a bug. First add the test and see it fail. Then implement the fix like you did here.

Are you familiar with testing? Usually a test has three parts: setup, execution and asserts.
In this case, for plotting, we currently have no asserts, but we just run the plot method once. This doesn't validate that it gives a good plot, but it does check cases like this one where code was changed and the plot method broke and we didn't notice. If there's a test that executes the code, then it fails if an error is raised.

The command to run the tests is
```
python setup.py  test -V
```
Or to run all the tests in a given file:
```
python setup.py test -V -t gammapy/data/tests/test_event_list.py
```
You can also generate a coverage report, showing which lines aren't tested at all:
```
python setup.py test -V -t gammapy/data/tests/test_event_list.py --coverage
open htmlcov/index.html
```

The coverage report from gammapy master is even available online - in this case you can see that the `EventList.plot_energy_hist` method isn't tested at all:
https://coveralls.io/builds/13810111/source?filename=gammapy%2Fdata%2Fevent_list.py#L530

**Comment by cosimoNigro** [comment]
Hi @cdeil,
I am not familiar with testing. What I did, after the modification, was to run - following your instructions into another thread:
`python -m pytest -v gammapy/irf/tests/test_energy_dispersion.py`
And I got all the test passed.

Although now I did other modifications (not committed) in the `event_list` class, what should I do?
We can also cancel this PR and wait for the modification of event_list mentioned in issue #1177.

Let me know

**Comment by cdeil** [comment]
@cosimoNigro - The commands in my last comment should get you started on testing with Gammapy.

We don't have much developer docs, but we do everything the same as in Astropy, and they have very extensive developer docs, so you can learn here:
http://astropy.readthedocs.io/en/stable/development/workflow/development_workflow.html#the-editing-workflow
http://astropy.readthedocs.io/en/stable/development/testguide.html#testing-guidelines

I learned pytest by reading / executing / playing with this tutorial - I think it's nice:
https://github.com/jiffyclub/pytest-features

Of course, let me know if you have any questions or issues.

---

It's up to you if you want to continue with this branch, or start a new one. Just let me know via comments when I should take action and what the status / plan is, like "this is ready from my side, please review" or "please merge, I plan to continue in another branch" or "I plan to continue working on this in the coming days".

File: gammapy/data/event_list.py [hunk 1]
```diff
@@ -532,11 +532,11 @@ def plot_energy_hist(self, ax=None, ebounds=None, **kwargs):
         from ..spectrum import CountsSpectrum
 
         if ebounds is None:
-            emin = np.min(self['ENERGY'].quantity)
-            emax = np.max(self['ENERGY'].quantity)
+            emin = np.min(self.energy.value) * self.energy.unit
```
Comment (cdeil) [inline gammapy/data/event_list.py]: Can you just change to this here?
```
emin = np.min(self.energy)
```
or
```
emin = self.energy.min()
```
?

I think the np.min function works fine with Quantity objects, no? At least this works for me:
```
import astropy.units as u
import numpy as np
np.min([1,2] * u.m)
```

**Comment by cosimoNigro** [comment]
Hi @cdeil,

I am modifying the EventList class according to what we discuss in Issue #1177.
I will open a new PR including this modification when I am done.

You are right `np.min/np.max` handles astropy units,
the correct modification should be
`self['ENERGY'].quantity` -> `self.table['ENERGY'].quantity`
the access to the `table` attribute was missing, this was triggering the issue. 

Is it ok for you if we close this one and I make this modification in the new PR I'll open when I finish with EventList class?

**Comment by cosimoNigro** [comment]
Closed, improvements included in the more general PR #1185

**Comment by cdeil** [comment]
Sure, let's continue in the other branch / PR. I'll have a look now.

**Pull request closed without merging**